### ISSUE TITLE
Exiting.In function: 'ParsePAF'

### ISSUE THREAD

**Issue by jinshui0202** [isovic/racon#23]
This is the error information:
#5: Unexpected value found! Overlaps are not formatted in the PAF format. Exiting.In function: 'ParsePAF'.

And my paf file looks like:
m170110_073744_42228_c101136512550000001823249906131712_s1_p0/30/0_13974	13974	507	13859	+	utg000001l	88984	24891	38334	2073	13443	255	cm:i:182
m170110_073744_42228_c101136512550000001823249906131712_s1_p0/68/0_3300	3300	204	2272	+	utg000697l	314885	115884	118067	145	2183	255	cm:i:12
m170110_073744_42228_c101136512550000001823249906131712_s1_p0/163/0_4021	4021	22	3956	+	utg000002l	43991	38915	42826	660	3934	255	cm:i:66
m170110_073744_42228_c101136512550000001823249906131712_s1_p0/163/0_4021	4021	158	3980	+	utg000003l	16020	9065	12833	666	3822	255	cm:i:56
m170110_073744_42228_c101136512550000001823249906131712_s1_p0/163/0_4021	4021	115	3879	+	utg002031l	46243	42607	46223	536	3764	255	cm:i:47
m170110_073744_42228_c101136512550000001823249906131712_s1_p0/163/0_4021	4021	22	3879	+	utg002471l	38305	2907	6571	452	3857	255	cm:i:41
m170110_073744_42228_c101136512550000001823249906131712_s1_p0/163/0_4021	4021	158	3653	-	utg002789l	20197	5756	9105	357	3495	255	cm:i:28
m170110_073744_42228_c101136512550000001823249906131712_s1_p0/163/0_4021	4021	1373	3351	-	utg000573l	96945	84380	86308	165	1978	255	cm:i:16
m170110_073744_42228_c101136512550000001823249906131712_s1_p0/163/0_4021	4021	2462	3358	-	utg001664l	52850	1969	2871	58	902	255	cm:i:7
m170110_073744_42228_c101136512550000001823249906131712_s1_p0/163/0_4021	4021	695	1039	+	utg002221l	63915	21629	21945	64	344	255	cm:i:6
m170110_073744_42228_c101136512550000001823249906131712_s1_p0/163/0_4021	4021	669	892	-	utg000531l	192555	100182	100380	49	223	255	cm:i:5
m170110_073744_42228_c101136512550000001823249906131712_s1_p0/163/0_4021	4021	732	890	-	utg002451l	79420	30732	30873	47	158	255	cm:i:4
m170110_073744_42228_c101136512550000001823249906131712_s1_p0/194/0_5043	5043	503	4875	-	utg000004l	55874	37728	50373	557	12645	255	cm:i:50
m170110_073744_42228_c101136512550000001823249906131712_s1_p0/194/0_5043	5043	460	5041	-	utg002719l	29432	101	5867	496	5766	255	cm:i:45
m170110_073744_42228_c101136512550000001823249906131712_s1_p0/194/0_5043	5043	387	5016	-	utg000005l	35283	160	7127	424	6967	255	cm:i:38
m170110_073744_42228_c101136512550000001823249906131712_s1_p0/194/0_5043	5043	283	2987	+	utg000004l	55874	37699	45545	278	7846	255	cm:i:26

**Comment by isovic** [comment]
Hi,
the only reason this error would happen is if a line could not be fully loaded here:
https://github.com/isovic/racon/blob/master/src/mhap.h#L52-L54
This doesn't include empty lines, because they are removed here:
https://github.com/isovic/racon/blob/master/src/mhap.cc#L206
Are you sure your PAF is formatted correctly?

Best regards,
Ivan.

**Comment by jinshui0202** [comment]
The PAF file was created by minimap as described in the racon README.md file like this:
minimap -t 24 reads.gfa.fasta all.reads.fq > mappings.paf

**Comment by jinshui0202** [comment]
Hello Ivan,
Thanks a lot for your response. Do find some error in the PAF file I parsed above?

**Comment by rvaser** [comment]
Hello,
the snip of your *.paf file looks fine to me. Did minimap finish without errors? Can you check the end of your *.paf file with 'tail -n 15'?

Best regards,
Robert

**Comment by PedroBarbosa** [comment]
Hello,

I'm having additional issues parsing paf files, as I see racon expects a cm column in the end of each line, but I don't have those, because I generated paf files from canu ovStoreDump utility.

My paf file looks like:

> 1         9473    1915    9192  -       2         7516     285    7503  4294966285        7277     255
1         9473     137    1608  +       106914    4602    2862    4280  4294967098        1471     255
1         9473    6407    6469  +       603386    3919    2075    3157  4294967290          62     255
1         9473    2437    6293  +       968119    3849     128    3836  4294966626        3856     255
1         9473       0    8901  +       1084988  11309    2700   11296  4294966112        8901     255
1         9473      12    8606  -       1084989  11201     275    8664  4294966130        8594     255
1         9473     339    8713  +       1084990  10999    2919   10986  4294966213        8374     255
1         9473    5450    6141  -       1318683   5442     141     813  4294967203         691     255
1         9473      60    1772  +       1647386   8047    6306    7965  4294967061        1712     255
1         9473      12    1758  -       1647389   6763      94    1813  4294967052        1746     255
> 

Does racon really use this information, or may I just create one additional column with random info written there?

Could you update the tool to accept 12 column TAB-delimited files, as Heng Li describes [here](https://github.com/lh3/miniasm/blob/master/PAF.md) ?

Thanks in advance,
Pedro

**Comment by rvaser** [comment]
Hello Pedro,
racon consensus mode filters reads by the number of shared minimizers in order to obtain a single mapping per read. Unfortunately, number of shared minimizers is in the 13th field of the PAF format which is an optional field in the format specification but minimap does produce it. Can the canu utility generate MHAP format instead of PAF?

On the other hand, if you are using racon in the error correction mode, all valid mappings are used for each read. You can add a dummy column in format "cm:i:<number>" to your PAF file, because it will not be used in this mode.

Sorry for the delayed answer and best regards,
Robert

P.s. In your file snip the 10th column seems to have quite large numbers (the column should contain equal bases in the mapping which should be smaller then the read lengths). You should double check if you run the utility or other programs as intended.

**Comment by PedroBarbosa** [comment]
Hello Robert,

I did try to generate MHAP output from the ovStoreDump utility, but didn't deepen on that (it should be possible, as overlaps were done with MHAP within canu). I preferred to add an extra column in the paf (used cm:i:15).  I'm running in the error correction mode. Paf file is 433Gb big (I replaced read IDs with numbers) and it's still on the parsing PAF stage (more than 17hours). 560Gb of RAM used, let's see if It will finish.

I'll take a look on the 10th column. Does racon need that column in the --erc mode? 

Best,
Pedro
